### PR TITLE
release notes for 1.38.1

### PR DESCRIPTION
**Pull request by jmazzitelli**
also removes the notes from the releases whose docs we recently removed

netlify link: https://deploy-preview-400--kiali.netlify.app/news/release-notes/

### CODE REVIEW

File: content/news/release-notes.adoc [hunk 1]
```diff
@@ -348,153 +360,6 @@ Fixes:
 * https://github.com/kiali/kiali/issues/3552[Kiali show service entries in another namespaces]
 * https://github.com/kiali/kiali/issues/3305["View in Grafana" link can be wrong]
 
-== 1.28.0
```
Comment (nrfox) [inline content/news/release-notes.adoc:351]: Why remove these? Is there another public changelog somewhere?

Is there a way for someone who is running one of the removed versions to still access the docs for that version? I'm guessing `git checkout kiali.io@<version.number>` and running `kiali.io` locally?

**Review by nrfox — APPROVED** [review]
One question but non-blocking

File: content/news/release-notes.adoc (hunk 1 above)
Comment (jmazzitelli) [reply content/news/release-notes.adoc:351]: These release notes are for the versions we have [currently documented on kiali.io](https://kiali.io/documentation/). See the [commit that recently removed the obsolete versions](https://github.com/kiali/kiali.io/pull/399/commits/f6144d545860caec72bf1f8df887a848d7864e9a). The commit comment explains, `Removing old kiali version: supporting only versions compatible with supported Istio versions`

Look at the [support matrix](https://kiali.io/documentation/latest/installation-guide/#_version_compatibility) to see the Kiali releases that currently support those upstream Istio releases that are supported (today that is Istio 1.9 and 1.10).

So, in short, we periodically clean out the docs and release notes so they include only those Kiali versions that work with supported versions of Istio.

If someone wants to see an old set of release notes, they could go to the release notes page in a specific branch in github, such as https://github.com/kiali/kiali.io/blob/v1.27.0/content/news/release-notes.adoc
